runtime/interpreter/interpreter.go

```diff
@@ -3369,15 +3382,17 @@ func (interpreter *Interpreter) authAccountCopyFunction(addressValue AddressValu
 
 func (interpreter *Interpreter) authAccountReadFunction(addressValue AddressValue, clear bool) *HostFunctionValue {
 
+	address := addressValue.ToAddress()
```

Comment:
Converted addresses can be cached and don't have to be recomputed on each function invocation

Reply:
Good idea! Refactored the remaining cases and added comments everywhere in fa1b621875708ed5c95cbd028cc4a56d2438eb97